File: paleo_workbench/ui/page_placeholder.py
from __future__ import annotations

from PySide6.QtCore import Qt
from PySide6.QtWidgets import QLabel, QVBoxLayout, QWidget

from paleo_workbench.ui import style


class PagePlaceholder(QWidget):
    def __init__(self, page_name: str, parent=None):
        super().__init__(parent)
        self.setObjectName("PagePlaceholder")
        self.name_label = QLabel(f"{page_name}\n(占位页, 待实现)")
        self.name_label.setAlignment(Qt.AlignmentFlag.AlignCenter)
        # style.bind：占位页文字色随主题刷新（此前构造时快照 light 值）
        style.bind(
            self.name_label,
            lambda: f"color: {style.palette()['TEXT_SECONDARY']}; font-size: 16px;",
        )
        layout = QVBoxLayout(self)
        layout.setContentsMargins(0, 0, 0, 0)
        layout.addStretch()
        layout.addWidget(self.name_label)
        layout.addStretch()

File: libs/ui_shell/src/page_placeholder.cpp
#include "pwb/ui_shell/page_placeholder.hpp"

#include <QVBoxLayout>

#include <pwb/ui_shell/style_registry.hpp>

namespace pwb::ui_shell {

PagePlaceholder::PagePlaceholder(const QString& page_name, QWidget* parent)
    : QWidget(parent) {
    setObjectName(QStringLiteral("PagePlaceholder"));
    name_label_ = new QLabel(page_name + QStringLiteral("\n(占位页, 待实现)"), this);
    name_label_->setAlignment(Qt::AlignmentFlag::AlignCenter);
    // style_bind: the label text color refreshes with the theme (the
    // Python original snapshot the light value at construction time until
    // style.bind fixed it).
    style_bind(name_label_, [] {
        const auto pal = style_palette();
        const auto it = pal.find("TEXT_SECONDARY");
        const QString color = it == pal.end()
                                  ? QStringLiteral("#666666")
                                  : QString::fromStdString(it->second);
        return QStringLiteral("color: %1; font-size: 16px;").arg(color);
    });
    auto* layout = new QVBoxLayout(this);
    layout->setContentsMargins(0, 0, 0, 0);
    layout->addStretch();
    layout->addWidget(name_label_);
    layout->addStretch();
}

}  // namespace pwb::ui_shell
